### PR TITLE
fix(dedicated.server): commitment feature not showing cancel options

### PR DESCRIPTION
**Pull request by stif59100**
ref:DTRSD-83000

Signed-off-by: stif59100 <steeve.vanderstocken@ovhcloud.com>

| Question         | Answer
| ---------------- | ---
| Branch?          | `release/components-w7`
| Bug fix?         | yes
| New feature?     | no
| Breaking change? | no
| Tickets          | DTRSD-83000
| License          | BSD 3-Clause

- [x] Try to keep pull requests small so they can be easily reviewed.
- [x] Commits are signed-off
- [ ] ~~Only FR translations have been updated~~
- [x] Branch is up-to-date with target branch
- [x] Lint has passed locally
- [x] Standalone app was ran and tested locally
- [x] Ticket reference is mentioned in linked commits (internal only)
- [ ] ~~Breaking change is mentioned in relevant commits~~

## Description

I corrected commitment feature not showing "Cancel" options

### CODE REVIEW

**Review by ghyenne — APPROVED** [review]

**Review by anooparveti — CHANGES_REQUESTED** [review]
**question:** I dont see all the changes required for the ticket. If server does not have engagementDetails, we have to call an `serviceInfos` API.

File: packages/manager/modules/billing-components/src/components/utils/billing.service.js [hunk 1]
```diff
@@ -46,6 +46,12 @@ export default class BillingService {
       );
   }
 
+  putServiceInfos(servicePath) {
+    return this.$http
+      .put(`${servicePath}/serviceInfos`)
```
Comment (anooparveti) [inline packages/manager/modules/billing-components/src/components/utils/billing.service.js:51]: **nitpick:** Pass `deleteAtExpiration:true` with the PUT call

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/billing-components/src/components/resiliation/resiliation.controller.js [hunk 2]
```diff
@@ -12,6 +12,11 @@ export default class BillingResiliationController {
       type: 'action',
     });
     this.isResiliating = true;
+    if (this.endStrategy === 'CANCEL' && !this.engagementDetails) {
+      return this.BillingService.putServiceInfos(
+        this.service.renew.deleteAtExpiration === true,
```
Comment (anooparveti) [inline packages/manager/modules/billing-components/src/components/resiliation/resiliation.controller.js:17]:
```suggestion
        this.service.id,
        { deleteAtExpiration:true }
```

File: packages/manager/modules/billing-components/src/components/utils/billing.service.js (hunk 1 above)
Comment (anooparveti) [inline packages/manager/modules/billing-components/src/components/utils/billing.service.js:51]:
```suggestion
  putServiceInfos(servicePath, params) {
    return this.$http
      .put(`${servicePath}/serviceInfos`, params)
```

**Review by anooparveti — CHANGES_REQUESTED** [review]

File: packages/manager/modules/billing-components/src/components/resiliation/resiliation.controller.js [hunk 3]
```diff
@@ -7,11 +7,17 @@ export default class BillingResiliationController {
   }
 
   resiliate() {
+    console.log(this.serviceName);
```
Comment (anooparveti) [inline packages/manager/modules/billing-components/src/components/resiliation/resiliation.controller.js:10]: **nitpick:** Remove the console statement

**Review by anooparveti — CHANGES_REQUESTED** [review]

**Review by anooparveti — APPROVED** [review]

**Review by ghyenne — APPROVED** [review]

**Pull request merged**